src/requestHandlers/hook.ts

```diff
@@ -0,0 +1,150 @@
+ * @param {Request} req 
+ * @param {Response} res 
+ * @returns 
+ */
+const createHook = async (req: Request, res: Response): Promise<Response> => {
+    const position : number[] = req.body.position;
+    const current_panel_id : number = req.body.current_panel_id;
+    const next_panel_set_id : number = req.body.next_panel_set_id;
+
+    const validArgs = assertArguments(
+        { position, current_panel_id, next_panel_set_id },
+        a => a != undefined || a === null,
+        'position or current_panel_id cannot be undefined'
+    );
+    if (!validArgs.success) return res.status(400).json(validArgs);
+
+    const response = await _createHookController(sequelize)(position, current_panel_id, next_panel_set_id);
+
+    return sanitizeResponse(response, res);
+};
+
+/**
+ * Get a hook from its id
+ * @param {number} id Hook's id
+ * @returns response or error
+ */
+const _getHookController = (sequelize: Sequelize) => async (id: number) => {
+    try {
+        return await hookService.getHook(sequelize)(id);
+    }
+    catch (err) {
+        return err;
+    }
+};
+
+/**
+ * Get Hook request
+ * @param {Request} req 
+ * @param {Response} res 
+ * @returns 
+ */
+const getHook = async (req: Request, res: Response): Promise<Response> => {
+    const id = req.body.id;
+
+    const validArgs = assertArgumentsDefined({ id });
+    if (!validArgs.success) return res.status(400).json(validArgs);
+
+    const response = await _getHookController(sequelize)(id);
+
+    return sanitizeResponse(response, res, `could not find hook with id ${req.body.id}`);
+};
+
+/**
+ * Get all hooks associated with a panel
+ * @param {number} id ID of target panel
+ * @returns response or error
+ */
+const _getPanelHooksController = (sequelize: Sequelize) => async (id: number) => {
+    try {
+        const panel = await getPanel(sequelize)(id);
+        if (panel == null) throw new Error('no panel exists for given panel id');
+        return await hookService.getPanelHooks(sequelize)(id);
+    }
+    catch (err) {
+        return err;
+    }
+};
+
+/**
+ * Get Panel Hooks request
+ * @param {Request} req 
+ * @param {Response} res 
+ * @returns 
+ */
+const getPanelHooks = async (req: Request, res: Response): Promise<Response> => {
+    const id = req.body.id;
+
+    const validArgs = assertArgumentsDefined({ id });
+    if (!validArgs.success) return res.status(400).json(validArgs);
+
+    const response = await _getPanelHooksController(sequelize)(id);
+
+    return sanitizeResponse(response, res, `could not find hooks under panel with id ${req.body.id}`);
+};
+
+/**
+ * Updates a hook with a next panel set
+ * @param {number} hook_id ID of hook that is being updated
+ * @param {number} panel_set_id ID of panel set that hook links to
+ * @returns response or error
+ */
+const _addSetToHookController = (sequelize: Sequelize) => async (hook_id: number, panel_set_id: number) => {
+    try {
+        const panelSet = await getPanelSetByID(panel_set_id);
```

Comment:
Needs to be updated with `(sequelize: Sequelize)` argument following the panel set PR (#6 ) model's approval